pandapower/create.py

```diff
@@ -3898,6 +3904,10 @@ def _create_column_and_set_value(net, index, variable, column, element, dtyp=flo
         net[element].at[index, column] = variable
     elif default_for_nan and column in net[element].columns:
         net[element].at[index, column] = default_val
+    try:
+        net[element][column] = net[element][column].astype(dtyp)
```

Comment:
I am wondering if the dtype of the new column will always be what the user expects... would it make more sense to leave the dtype out and try to somehow let pandas handle this?

Reply:
let's have a short call about that...

Reply:
I don't think that it is expected behaviour that it is not tried to set the dtype which is passed to `_create_column_and_set_value()`. But I am adding a `if dtype is not None` clause to allow pandas to handle that issue if wished.